shtdlib.sh

```diff
@@ -2227,6 +2223,54 @@ function test_shopt_decorator {
     assert shopt -qo pipefail && color_echo green "Successfully decorated  ${FUNCNAME[0]} with pipefail"
 }
 
+# Test signaling
+function test_signal_process {
+    signal_processor SIGUSR2 'exit 42' > /dev/null
+    local sub_pid_0="${!}"
+    signal_processor SIGUSR1 "sleep 2 && kill -s SIGUSR2 ${sub_pid_0} && exit 42" > /dev/null
+    local sub_pid_1="${!}"
+    debug 10 "Spawned sub processes using signal processor with pids: ${sub_pid_0} and ${sub_pid_1}"
+    debug 10 "Active sub processes are: $(pgrep -P ${$} | tr '\n' ' ')"
+    signal_process "${sub_pid_1}" SIGUSR1 > /dev/null
+    debug 10 "Waiting for sub processes to exit"
+    bash -c "sleep 10 && kill ${sub_pid_0} &> /dev/null" &
```

Comment:
Are these sleep/kills a safety net?

Reply:
In case of failure we'll never actually return without this, so it's a grace period in case of failure so that the tests will return and fail yes.